bindings/C/adios2/c/adios2_c_types.h

```diff
@@ -180,6 +180,13 @@ typedef struct
     adios2_blockinfo *BlocksInfo;
 } adios2_varinfo;
 
+typedef struct
+{
+    int major;
+    int minor;
+    int patch;
+} adios2_version_struct;
```

Comment:
```suggestion
    int patch;
    char str[sizeof(ADIOS2_VERSION_STR)];
} adios2_version_struct;
```

Reply:
That will not work. This struct has a size that is not known in advance, and it is impossible to call it in a dynamic library. Almost every new version of ADIOS2 will have a different size for this struct, leading to ABI incompatibilities.

Why not use a `const char* str` field that points to a constant string that does not need to be deallocated?